crates/miden-objects/src/note/note_tag.rs

```diff
@@ -181,41 +199,46 @@ impl NoteTag {
             return Err(NoteError::NoteTagUseCaseTooLarge(use_case_id));
         }
 
-        let execution_bits = LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED;
         let use_case_bits = (use_case_id as u32) << 16;
         let payload_bits = payload as u32;
 
-        Ok(Self(execution_bits | use_case_bits | payload_bits))
+        Ok(Self::LocalAccount(0b10 | use_case_bits | payload_bits))
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns true if the note is intended for execution by a specific account.
+    /// Returns true if the note is intended for execution by a specific account.(network or local)
     ///
-    /// A note is intended for execution by a single account if the first two bits are zeros
+    /// A note is intended for execution by a single account if the first two bits are `0b00` or
+    /// `0b10`
     pub fn is_single_target(&self) -> bool {
-        let first_2_bit = self.0 >> 30;
-        first_2_bit == 0b00
+        matches!(self, NoteTag::NetworkAccount(_) | NoteTag::LocalAccount(_))
     }
 
     /// Returns note execution mode defined by this tag.
     ///
     /// If the most significant bit of the tag is 0 the note is intended for local execution;
     /// otherwise, the note is intended for network execution.
     pub fn execution_mode(&self) -> NoteExecutionMode {
-        let first_bit = self.0 >> 31;
-
-        if first_bit == (LOCAL_EXECUTION as u32) {
-            NoteExecutionMode::Local
-        } else {
-            NoteExecutionMode::Network
+        match self {
+            NoteTag::NetworkAccount(_) | NoteTag::NetworkUseCase(..) => NoteExecutionMode::Network,
+            NoteTag::LocalAccount(_) | NoteTag::LocalUseCase(..) => NoteExecutionMode::Local,
         }
     }
 
     /// Returns the inner u32 value of this tag.
-    pub fn inner(&self) -> u32 {
-        self.0
+    pub fn as_u32(&self) -> u32 {
+        match self {
+            NoteTag::NetworkAccount(tag) => *tag,
+            NoteTag::NetworkUseCase(use_case_bits, payload_bits) => {
+                NETWORK_USECASE | *use_case_bits as u32 | *payload_bits as u32
+            },
+            NoteTag::LocalAccount(tag) => LOCAL_ACCOUNT | *tag,
```

Comment:
```suggestion
            NoteTag::LocalAccount(tag) => *tag,
```

This shouldn't be needed if we include the prefix directly in the constructors.